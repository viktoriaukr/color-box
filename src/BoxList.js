import React, { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

const BoxList = () => {
  const [boxes, setBoxes] = useState([]);
  const addBox = (color) => {
    setBoxes((boxes) => [...boxes, color]);
  };
  const remove = (id) =>
    setBoxes((boxes) => boxes.filter((box) => box.id !== id));
  const newBox = boxes.map((box) => (
    <Box
      key={box.id}
      id={box.id}
      width={box.width}
      height={box.height}
      backgroundColor={box.backgroundColor}
      remove={remove}
    />
  ));
  return (
    <div>
      <NewBoxForm addBox={addBox} />
      <br />
      {newBox}
    </div>
  );
};

export default BoxList;
