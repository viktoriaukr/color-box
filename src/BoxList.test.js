import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

function addBox(box, width = "5", height = "5", color = "red") {
  const boxWidth = box.getByLabelText("Width");
  const boxHeight = box.getByLabelText("Height");
  const boxColor = box.getByLabelText("Background Color");
  fireEvent.change(boxWidth, { target: { value: width } });
  fireEvent.change(boxHeight, { target: { value: height } });
  fireEvent.change(boxColor, { target: { value: color } });
  const button = box.getByText("Add Box");
  fireEvent.click(button);
}

test("render", () => {
  render(<BoxList />);
});

test("matches snapshot", () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

test("adding a box", () => {
  const box = render(<BoxList />);
  expect(box.queryByText("X")).not.toBeInTheDocument();
  addBox(box);
  expect(box.queryByText("X")).toBeInTheDocument();
  expect(box.getAllByDisplayValue("")).toHaveLength(3);
});

test("removing box", () => {
  const box = render(<BoxList />);
  addBox(box);
  const btn = box.getByText("X");
  fireEvent.click(btn);
  expect(btn).not.toBeInTheDocument();
});
