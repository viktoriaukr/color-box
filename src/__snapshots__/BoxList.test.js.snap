// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        name="width"
        placeholder="Width"
        type="text"
        value=""
      />
      <br />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        name="height"
        placeholder="Height"
        type="text"
        value=""
      />
      <br />
      <label
        for="backgroundColor"
      >
        Background Color
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        placeholder="Background Color"
        type="text"
        value=""
      />
      <br />
      <button>
        Add Box
      </button>
    </form>
    <br />
  </div>
</DocumentFragment>
`;
